src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -1535,7 +1535,7 @@ void ShenandoahFullGC::phase5_epilog() {
         heap->generation_sizer()->transfer_to_young(excess_old_regions);
       } else if (old_capacity < old_usage) {
         size_t old_regions_deficit = (old_usage - old_capacity) / ShenandoahHeapRegion::region_size_bytes();
-        heap->generation_sizer()->transfer_to_old(old_regions_deficit);
+        heap->generation_sizer()->force_transfer_to_old(old_regions_deficit);
       }
```

Comment:
So here, we may end up violating max/min sizes of young generation.  I'm inclined to allow this if that is required to make the just-completed full gc valid.

Reply:
We may violate a configured constraint, but the usage accounting should be correct now.

Reply:
good.  and i assume that the configured constraint violation no longer manifests as a failed assertion.